app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,37 @@
+class VotesController < ApplicationController
+
+  def create
+    if params[:answer_id]
+      @votable = Answer.find(params[:answer_id])
+    else
+      @votable = Question.find(params[:question_id])
+    end
+
+    if !@votable.voted.include?(current_user.id)
+      @vote = @votable.votes.build(value: params[:value], voter: current_user)
+      if @vote.save
+        if request.xhr?
+          render :json => {votable_id: @votable.id, votable_vote_count: @votable.vote_count}
+        else
+          if params[:answer_id]
+            redirect_to question_path(@vote.votable.question_id)
+          else
+            redirect_to question_path(@vote.votable)
+          end
+        end
+      else
+        if params[:answer_id]
+           render question_path(@vote.votable.question_id)
+        else
+           render question_path(@vote.votable)
+        end
```

Comment:
Give the user some feedback if the save fails. Also, it might be worth pulling out xhr vs not into two private methods so that your create action is more readable.